src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/summaries/temperature/CompactNodeSummary.java

```diff
@@ -68,6 +74,69 @@ public CompactNodeSummary(final String nodeId, final String hostAddress) {
         this.numOfShards = new int[TemperatureVector.Dimension.values().length];
     }
 
+    public static CompactNodeSummary buildSummaryFromDatabase(Result<Record> records,
+        DSLContext context) {
+        if (records.size() != 1) {
+            throw new IllegalArgumentException(
+                "Only 1 CompactNodeSummary expected. Found: " + records.size());
```

Comment:
Should we also add to the log, what they were ? Basically add `records` object to the log ?

Reply:
Done.